calculator.c

```diff
@@ -1,128 +1,93 @@
-#include <stdio.h>
-#include <stdlib.h>
-#include <ctype.h>
-#include <string.h>
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
 
-#define MAX 100
-
-int eval_exp(const char* exp, int* flag);
-int precedence(char operator);
-int calc_exp(int a, int b, char operator);
-
-int main() 
+int is_dig(char c)
 {
-    char exp[MAX];
-
-    printf("Enter a math expression: ");
-    fgets(exp, MAX, stdin);
-    exp[strcspn(exp, "\n")] = '\0';
-
-    int flag = 0;
-    int result = eval_exp(exp, &flag);
+    if(c>='0' && c<='9')
+        return 1;
+    else
+        return 0;
+}
 
-    if (flag == 1) 
-    {
-        printf("Error: Division by zero.\n");
-    } 
-    else if (flag == 2) 
-    {
-        printf("Error: Invalid expresiion.\n");
-    } 
-    else 
+int calc_exp(int x, int y, char op)
+{
+    switch(op)
     {
-        printf("Result: %d\n", result);
+        case '+': return x+y;
+        case '-': return x-y;
+        case '*': return x*y;
+        case '/': 
+        if(y!=0)
+            return x/y;
+        else
+        {
+            printf("Can't divide by zero");
+            exit(1);
+        }
     }
-
-    return 0;
 }
 
-int eval_exp(const char* exp, int* flag) 
+int evaluate_exp(char exp[])
 {
-    int values[MAX];      
-    char operators[MAX];   
-    int top = -1;    
-    int op_top = -1;  
+    int numstack[100], top=0;
+    char opstack[100], e;
+    int optop=0;
+    int num=0, result;
 
-    for (int i = 0; exp[i] != '\0'; i++) 
+    for(int i=0;exp[i]!='\0';i++)
     {
-        if (isspace(exp[i])) {
```

Comment:
trying implementing isspace and isdigit as it would be more helpful for learning purpose

Reply:
I have implemented isdigit function, it is named as is_dig 
I haven't used isspace function in my program

Reply:
I think this has raised on the old code not on the new one